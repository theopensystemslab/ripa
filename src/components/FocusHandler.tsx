import { makeStyles } from "@material-ui/core/styles";
import * as React from "react";

const useStyles = makeStyles({
  root: {
    "&:focus": {
      outline: "none"
    }
  }
});

const FocusHandler = ({ tabIndex = 0, children, ...props }) => {
  const container = React.useRef(null);
  const classes = useStyles();
  return (
    <div
      className={classes.root}
      tabIndex={tabIndex}
      {...props}
      ref={container}
    >
      {children}
    </div>
  );
};

export default FocusHandler;
