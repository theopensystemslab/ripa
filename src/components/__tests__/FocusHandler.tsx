import { render } from "@testing-library/react";
import * as React from "react";

import FocusHandler from "../FocusHandler";

describe("Focus Handler Component", () => {
  const children = "content goes here";
  it("should render snapshot", () => {
    const { asFragment } = render(<FocusHandler>{children}</FocusHandler>);
    expect(asFragment()).toMatchSnapshot();
  });
});
