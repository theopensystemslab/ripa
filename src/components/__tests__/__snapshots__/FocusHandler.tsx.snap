// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Focus Handler Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    tabindex="0"
  >
    content goes here
  </div>
</DocumentFragment>
`;
